### PR TITLE
make rate_limit redis keys expire whatever would happen

### LINKED ISSUE #980: rate_limit not atomic
It seems that sometimes rate_limit doesn't set TTL on newly created entries.

I've got a couple of cases of that and had to clean them up manually. Not having a TTL means user will certainly hit the rate limit. 

So, investigate the code and wrap it in a transaction.

Thanks.

### PR DESCRIPTION
**Pull request by celesteking**
Fix #980 : my proposal.

### CODE REVIEW

File: plugins/rate_limit.js [hunk 1]
```diff
@@ -150,26 +150,30 @@ exports.rate_limit = function (connection, key, value, cb) {
...
+                return cb(null, true);
+            }
+            else {
+                // OK
+                return cb(null, false);
+            }
+
+        };
+
+        if (val == null) // new key
```
Comment (smfreegard) [inline plugins/rate_limit.js]: Missing {'s

**Comment by celesteking** [comment]
okay, anything else? it works fine without {} there.

**Comment by smfreegard** [comment]
It works - but it's not the house style: https://github.com/baudehlo/Haraka/wiki/Contributing#style-conventions

Otherwise - looks good.

**Comment by celesteking** [comment]
Well here you go. :)

**Pull request merged**